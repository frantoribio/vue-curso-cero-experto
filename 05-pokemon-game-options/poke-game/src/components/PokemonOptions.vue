<template>
  <p>Pokemon Options</p>
</template>

<script>
  export default {
    name: 'PokemonOptions',
  }
</script>

<style lang="scss" scoped></style>
