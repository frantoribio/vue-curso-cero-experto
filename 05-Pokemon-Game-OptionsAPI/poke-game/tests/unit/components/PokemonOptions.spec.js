import { describe, test, expect, beforeEach } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions.vue'
import { pokemons } from '../mocks/pokemons.mocks'

describe('Pokemon Options Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: pokemons,
      },
    })
  })

  test('Debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Debe mostrar las cuatro opciones correctamente', () => {
    const liTags = wrapper.findAll('li')
    expect(liTags.length).toBe(4)
    expect(liTags.at(0).text()).toBe('bulbasaur')
    expect(liTags.at(1).text()).toBe('ivysaur')
    expect(liTags.at(2).text()).toBe('venusaur')
    expect(liTags.at(3).text()).toBe('charmander')
  })

  test('Debe emitir selection-pokemon con sus datos', () => {
    const [li1, li2, li3, li4] = wrapper.findAll('li')
    li1.trigger('click')
    li2.trigger('click')
    li3.trigger('click')
    li4.trigger('click')
    expect(wrapper.emitted('selection-pokemon').length).toBe(4)
    expect(wrapper.emitted('selection-pokemon')[0][0]).toStrictEqual(pokemons[0].id)
    expect(wrapper.emitted('selection-pokemon')[1][0]).toStrictEqual(pokemons[1].id)
    expect(wrapper.emitted('selection-pokemon')[2][0]).toStrictEqual(pokemons[2].id)
    expect(wrapper.emitted('selection-pokemon')[3][0]).toStrictEqual(pokemons[3].id)
  })
})
