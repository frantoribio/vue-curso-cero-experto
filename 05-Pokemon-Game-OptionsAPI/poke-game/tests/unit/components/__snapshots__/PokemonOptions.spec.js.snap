// Vitest Snapshot v1

exports[`Pokemon Options Component > Debe hacer match con el snapshot 1`] = `
"<div class=\\"options-container\\" data-v-39189cb9=\\"\\">
  <ul data-v-39189cb9=\\"\\">
    <li data-v-39189cb9=\\"\\">bulbasaur</li>
    <li data-v-39189cb9=\\"\\">ivysaur</li>
    <li data-v-39189cb9=\\"\\">venusaur</li>
    <li data-v-39189cb9=\\"\\">charmander</li>
  </ul>
</div>"
`;
